### ISSUE TITLE
Comparison operators `==`/`!=` reported too aggressively

### ISSUE THREAD

**Issue by jfoclpf** [jshint/jshint#3102]
I have a config.json file

```
{
  "eqeqeq": false
}
```

I typed

`jshint --config "/path/to/my/config.json"`

and still I get errors/warnings regarding "==" and "==="

What is wrong?

**Comment by jfoclpf** [comment]
Also I tryed a config.json

```
{
    "jshintConfig": {
        "eqeqeq": false,
        "evil": false,
        "multistr": false
    }
}
```
then I typed:

jshint --config "/path/to/my/config.json"

and still the linting options do not take effect.

Could anyone kindly help!? I'm working around for 1 day and I can't figure out what I am doing wrong.

**Comment by nicolo-ribaudo** [comment]
Does it work if you use `eqeqeq: true`?

**Comment by jfoclpf** [comment]
Nope, now as I type `jshint --config "/path/to/my/config.json"` I get `Error:SyntaxError: Unexpected token e`

**Comment by jugglinmike** [comment]
This is very strange. Could you share the input code, preferably reduced as
much as possible? Could you also let us know which version of JSHint you are
using (via `jshint --version`)?

**Comment by jfoclpf** [comment]
Sure

The version I'm using is v2.9.4

Here it goes the input code:
![capturar](https://cloud.githubusercontent.com/assets/3984909/23107304/e3fed270-f6fa-11e6-8657-8009132e7001.JPG)

**Comment by jugglinmike** [comment]
By "the input code," I meant, "the JavaScript program that you are linting."

**Comment by jfoclpf** [comment]
Sure, I attach here

The first error appears, as you can see on the above shell, on line 63, which is simply:

```
    if(length==0){
        return null;
    }
```

[myscript.zip](https://github.com/jshint/jshint/files/786321/myscript.zip)

**Comment by jfoclpf** [comment]
If you try this simple code, it gives the same warning

```
function test(input_var){
    if (input_var == 0){
        return true;
    }
    else{
        return false;
    }  
}
```

But it seems that it appears only while comparing with '0', because this code gives no error

```
function test(input_var){
    if (input_var == "1"){
        return true;
    }
    else{
        return false;
    }  
}
```

Is it supposed to be like that?

**Comment by jugglinmike** [comment]
The `eqeqeq` option is an "enforcing option," meaning that it should only
produce warnings when set to `true`. You have identified an inconsistency in
JSHint's enforcement of this particular linting rule.

The option's description states, "This options prohibits the use of == and !=
in favor of `===` and `!==`." Enabling it will cause JSHint to issue warning
W118 for offending code. However, JSHint will issue a separate warning (W041)
if either side of the comparison is a "falsey" literal value, *regardless of
the value of the `eqeqeq` option*.

**The long-term resolution for the project**, removing W041 altogether. It's a
shame because it makes JSHint less strict by default, but the current behavior
is inconsistent and confusing. And while the warning message for W041 is far
more descriptive (and therefore a better choice to keep around), we can't
change W118 in-place outside of a major version release. This should be
relatively straightforward, so I've marked this issue as a "good first bug."

**The short-term resolution for users** is to disable W041 in your project
configuration, either with an in-line directive:

```js
// jshint -W041
```

Or within a JSHint configuration file:

```js
{
  "-W041": true
}
```

**Comment by jfoclpf** [comment]
Thank you for your message

But your option `"-W041": true` still gives an error while comparing  == with 0

**Comment by jugglinmike** [comment]
Can you be more specific? For instance, if I paste the following code into the
editor on http://jshint.com, then I see the warning, 

```js
if (x == 0) {}
```

...then I see the following warning:

> One warning
>
> 1 Use '===' to compare with '0'.

However, when I insert that option, the warning is suppressed:

```js
// jshint -W041: true
if (x == 0) {}
```

How does your experience differ?

**Comment by jfoclpf** [comment]
I didn't try with the comment // in the .js file. 
I simply added this line `W041": true` to the JSON object of  the JSHint configuration file.

**Comment by jugglinmike** [comment]
In that case, we'll need to ensure that JSHint is actually reading that file. Can you try deliberately inserting a JSON syntax error--for instance:

```json
{
  "-W041": true,
  this is not valid JSON
}
```

If everything is configured properly, JSHint should report a syntax error in the configuration file. If it does not, then JSHint isn't reading that file at all.

**Comment by jfoclpf** [comment]
It works as you said, it gave an error

![capturar](https://cloud.githubusercontent.com/assets/3984909/23344537/bda4cf38-fc7e-11e6-8bc5-ba6102996c80.JPG)

**Comment by josephzidell** [comment]
Remove the trailing comma:

```json
{
  "-W041": true
}
```

**Comment by jfoclpf** [comment]
@josephzidell I was told not to by @jugglinmike so we can test the config file, check previous message

**Comment by jugglinmike** [comment]
I'm still unable to reproduce the error. This is very tricky. Since it seems to
require multiple files, I've created a standalone project on GitHub to help us
isolate the problem:

https://github.com/jshint/gh-3102

Can you clone that repository, follow the instructions, and let me know what
you find?

**Comment by jfoclpf** [comment]
I cloned and I tested.
Exactly as you said:

It seems it is working

![capturar](https://cloud.githubusercontent.com/assets/3984909/23834328/2bcdf418-0755-11e7-9638-c402106f62bc.JPG)

**Comment by jfoclpf** [comment]
It seems that it does not save the configuration file, because when I run just in one line, it works, but when I use in two command lines, it doesn't work

```
./node_modules/.bin/jshint --config cfg_jshint.json
./node_modules/.bin/jshint myscript.js
```

**Comment by jugglinmike** [comment]
Ah, yes--the command-line interface is stateless, so one invocation will not
effect the next (note that this may change in the future, though--[we recently
received a feature request for result caching](https://github.com/jshint/jshint/issues/3107)).

This issue has grown a bit unwieldy, making it a little confusing for use as a
bug report. I've created gh-3109 to track the bug and reverted this issue's
title to the original value.

Glad we got to the bottom of this one!
